datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -811,7 +813,7 @@ def _to_bytes(self, data):
         if not isinstance(data, bytes):
             try:
                 return data.encode('utf-8')
-            except Exception:
+            except UnicodeError:
```

Comment:
On the one hand, that makes sense, on the other hand, I wonder if that doesn't hide some errors we don't catch elsewhere... I'd be hesitant to make that change.

Reply:
Yes, like if data was a different type

Reply:
Which exception should we catch to be more permissive but still not as permissive as catching everything? 
We should add specific test for these as well I believe.

Reply:
Let's go back to Exception